examples/flight_replay/flight_replay.py

```diff
@@ -7,122 +7,178 @@
+from dronekit import connect, Command, VehicleMode, LocationGlobalRelative
 from pymavlink import mavutil
 import json, urllib, math
+import time
 
 #Set up option parsing to get connection string
 import argparse  
-parser = argparse.ArgumentParser(description='Get a flight from droneshare and send as waypoints to a vehicle. Connects to SITL on local PC by default.')
-parser.add_argument('--connect', default='127.0.0.1:14550',
-                   help="vehicle connection target. Default '127.0.0.1:14550'")
-parser.add_argument('--mission_id', default='101',
-                   help="DroneShare Mission ID to replay. Default is '101'")                   
-parser.add_argument('--api_server', default='https://api.3drobotics.com',
-                   help="API Server. Default is Droneshare API (https://api.3drobotics.com)")
-parser.add_argument('--api_key', default='89b511b1.d884d1cb57306e63925fcc07d032f2af',
-                   help="API Server Key (default should be fine!)")
+parser = argparse.ArgumentParser(description='Load a telemetry log and use position data to create mission waypoints for a vehicle. Connects to SITL on local PC by default.')
+parser.add_argument('--connect', help="vehicle connection target.")
+parser.add_argument('--tlog', default='flight.tlog',
+                   help="Telemetry log containing path to replay")
 args = parser.parse_args()
 
 
-# Connect to the Vehicle
-print 'Connecting to vehicle on: %s' % args.connect
-vehicle = connect(args.connect, wait_ready=True)
-
-
-
-def _decode_list(data):
-    """A utility function for decoding JSON strings from unicode"""
-    rv = []
-    for item in data:
-        if isinstance(item, unicode):
-            item = item.encode('utf-8')
-        elif isinstance(item, list):
-            item = _decode_list(item)
-        elif isinstance(item, dict):
-            item = _decode_dict(item)
-        rv.append(item)
-    return rv
-
-def _decode_dict(data):
-    """A utility function for decoding JSON strings from Unicode"""
-    rv = {}
-    for key, value in data.iteritems():
-        if isinstance(key, unicode):
-            key = key.encode('utf-8')
-        if isinstance(value, unicode):
-            value = value.encode('utf-8')
-        elif isinstance(value, list):
-            value = _decode_list(value)
-        elif isinstance(value, dict):
-            value = _decode_dict(value)
-        rv[key] = value
-    return rv
-
-def download_messages(mission_id, max_freq = 1.0):
-    """Download a public mission from droneshare (as JSON)"""
-    message_url="%s/api/v1/mission/%s/messages.json?max_freq=%s&api_key=%s" % (args.api_server, mission_id, max_freq, args.api_key)
-    f = urllib.urlopen(message_url)
-    j = json.load(f, object_hook=_decode_dict)
-    f.close()
-    return j
-
-def replay_mission(payload):
-    """Given mission JSON, set a series of wpts approximating the previous flight"""
+def start_default_sitl(lat=-35.363261, lon=149.165230):
```

Comment:
Gah!  I set it all up in the right order to pull the start location out of the first message - and then forgot to do that!